src/controller/java/AndroidCallbacks.h

```diff
@@ -138,6 +139,24 @@ struct InvokeCallback : public app::CommandSender::Callback
     JniGlobalReference mWrapperCallbackRef;
 };
 
+struct BatchInvokeCallback : public app::CommandSender::ExtendableCallback
+{
+    BatchInvokeCallback(jobject wrapperCallback);
+    ~BatchInvokeCallback();
+
+    void OnResponse(app::CommandSender * commandSender, const app::CommandSender::ResponseData & aResponseData) override;
+    void OnNoResponse(app::CommandSender * commandSender, const app::CommandSender::NoResponseData & aNoResponseData) override;
+    void OnError(const app::CommandSender * apCommandSender, const app::CommandSender::ErrorData & aErrorData) override;
+    void OnDone(app::CommandSender * apCommandSender) override;
+
+    CHIP_ERROR GetIndexForCommandRef(uint16_t aCommandRef, size_t & aIndex);
+    CHIP_ERROR AddCommandRefToIndexLookup(uint16_t aCommandRef, size_t aIndex);
+
+    std::unordered_map<uint16_t, size_t> commandRefToIndex;
```

Comment:
If you don't want to, you don't have to use an unordered_map here. I would argue that is adds complexity without any benefit. This is something I am meaning to clean up in chip-repl.

In your case, because you are providing the `commandRef` to be the index already. You don't need this at all. If you want to be very safe, you can store the length of the array and validate that the CommandRef you get back in the OnResponse is less than the max. But you can rely on that `CommandRef` being the index for you.

Reply:
Agree, would remove it, thanks

Reply:
removed, thanks